creator-node/scripts/updateContentBlacklist.js

```diff
@@ -1,128 +1,310 @@
+  let allSegments = await getSegments(type, values)
+
+  // Hit creator node /ipfs/:CID route to see if segment is blacklisted
+  let checkFn, filterFn
+  switch (action) {
+    case 'ADD':
+      checkFn = checkIsBlacklisted
+      filterFn = status => !status
+      break
+    case 'REMOVE':
+      checkFn = checkIsNotBlacklisted
+      filterFn = status => status
+      break
+  }
+
+  // Batch requests
+  let creatorNodeResponses = []
+  const checkSegmentBlacklistStatusRequests = allSegments.map(segment => checkFn(segment))
+  for (let i = 0; i < allSegments.length; i += REQUEST_CONCURRENCY_LIMIT) {
+    const creatorNodeResponsesSlice = await Promise.all(checkSegmentBlacklistStatusRequests.slice(i, i + REQUEST_CONCURRENCY_LIMIT))
+    creatorNodeResponses = creatorNodeResponses.concat(creatorNodeResponsesSlice)
+  }
+
+  // Segments that were not accounted for during blacklisting/unblacklisting
+  const unaccountedSegments = creatorNodeResponses
+    .filter(resp => filterFn(resp.blacklisted))
+    .map(resp => resp.segment)
+
+  if (unaccountedSegments.length > 0) {
+    let errorMsg = `Some segments from [${type}] and values [${values}] were not blacklisted/unblacklisted.`
+    errorMsg += `\nNumber of Segments: ${unaccountedSegments.length}`
+    errorMsg += `\nSegments: [${unaccountedSegments.toString()}]`
+    throw new Error(errorMsg)
+  }
+
+  return allSegments
+}
+
+async function getSegments (type, values) {
+  if (type === 'CID') return values
+
+  let allSegments = []
+  let discProvRequests
+  try {
+    // Fetch the segments via disc prov
+    switch (type) {
+      case 'USER': {
+        discProvRequests = values.map(value => axios({
+          url: `${DISCOVERY_PROVIDER_ENDPOINT}/tracks`,
+          method: 'get',
+          params: { user_id: value },
+          responseType: 'json'
+        }))
+        break
+      }
+      case 'TRACK': {
+        discProvRequests = values.map(value => axios({
+          url: `${DISCOVERY_PROVIDER_ENDPOINT}/tracks`,
+          method: 'get',
+          params: { id: value },
+          responseType: 'json'
+        }))
+        break
+      }
+    }
+
+    // Batch requests
+    let discProvResps = []
+    for (let i = 0; i < discProvRequests.length; i += REQUEST_CONCURRENCY_LIMIT) {
+      const discProvResponsesSlice = await Promise.all(discProvRequests.slice(i, i + REQUEST_CONCURRENCY_LIMIT))
+      discProvResps = discProvResps.concat(discProvResponsesSlice)
+    }
+
+    for (const resp of discProvResps) {
+      for (const track of resp.data.data) {
+        for (const segment of track.track_segments) {
+          allSegments.push(segment.multihash)
+        }
+      }
+    }
+  } catch (e) {
+    throw new Error(`Error with fetching segments for verifcation: ${e}`)
+  }
+  return allSegments
+}
+
+async function checkIsBlacklisted (segment) {
+  try {
+    await axios({
+      url: `${CREATOR_NODE_ENDPOINT}/ipfs/${segment}`,
+      method: 'get',
```

Comment:
could this be a head request?

Reply:
hmmm yes. but since we return a 403 before it streams anything i guess it doesn't really matter. can update!